### PR TITLE
Update EOL versions

### PR DESCRIPTION
**Pull request by richlander**

### CODE REVIEW

**Review by jjonescz — APPROVED** [review]

File: azure-pipelines.yml [hunk 1]
```diff
@@ -148,23 +148,23 @@ stages:
   # Like template `eng/common/templates/jobs/source-build.yml`
   - job: Source_Build_Managed
     displayName: Source-Build (Managed)
-    container: mcr.microsoft.com/dotnet-buildtools/prereqs:centos-stream8
+    container: mcr.microsoft.com/dotnet-buildtools/prereqs:cbl-mariner-2.0
     pool:
       ${{ if eq(variables['System.TeamProject'], 'public') }}:
         name: $[replace(replace(eq(contains(coalesce(variables['System.PullRequest.TargetBranch'], variables['Build.SourceBranch'], 'refs/heads/main'), 'release'), 'true'), True, 'NetCore-Svc-Public' ), False, 'NetCore-Public')]
-        demands: ImageOverride -equals Build.Ubuntu.1804.Amd64.Open
+        demands: ImageOverride -equals Build.Ubuntu.2204.Amd64.Open
```
[the remark this answers is not in the document]
Comment (richlander) [reply azure-pipelines.yml:155]: The intended pattern is:

- For raw VM usage, use oldest supported Ubuntu
- For container-based testing, use the newest

File: azure-pipelines.yml [hunk 2]
```diff
@@ -148,23 +148,23 @@ stages:
...
+        demands: ImageOverride -equals Build.Ubuntu.2204.Amd64
     workspace:
       clean: all
     steps:
     - template: /eng/common/templates/steps/source-build.yml
       parameters:
         platform:
           name: 'Managed'
-          container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:centos-stream8'
+          container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:cbl-mariner-2.0'
```
Comment (MichaelSimons) [inline azure-pipelines.yml:167]: Source build legs should continue to use centos as we don't ship a SB product via mariner.

```suggestion
          container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:centos-stream9'
```

File: azure-pipelines.yml [hunk 3]
```diff
@@ -148,23 +148,23 @@ stages:
   # Like template `eng/common/templates/jobs/source-build.yml`
   - job: Source_Build_Managed
     displayName: Source-Build (Managed)
-    container: mcr.microsoft.com/dotnet-buildtools/prereqs:centos-stream8
+    container: mcr.microsoft.com/dotnet-buildtools/prereqs:cbl-mariner-2.0
```
Comment (MichaelSimons) [inline azure-pipelines.yml:151]: Source build legs should continue to use centos as we don't ship a SB product via mariner.

```suggestion
      container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:centos-stream9'
```

**Review by jaredpar — APPROVED** [review]

**Review by MichaelSimons — APPROVED** [review]
SB changes LGTM

**Pull request merged**